src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java

```diff
@@ -235,6 +242,17 @@ public static JsonObject flagToJson(final CheckFlag flag,
                 .map(tag -> String.format("%s=%s", HighwayTag.KEY, tag.getTagValue()));
     }
 
+    private static JsonObject getFixSuggestionDescriptions(final CheckFlag flag)
+    {
+        final JsonObject fixSuggestionObject = new JsonObject();
```

Comment:
I thought it might be more convenient as an object, so that it would be easy to look up suggestions for the flagged object and you can still easily loop through it. 
As for the OSM ID, we could add it if it seems like that would really be helpful. However, it is quite a simple calculation to get it from the Atlas ID, and realistically it should exist in the properties of one of the flagged objects.

Reply:
Yeah, I can see how that would be helpful for looks ups. My only concern is that it wouldn't follow the convention for all other collection type objects we store in JSON.
Also.. I think it would be useful if he added it to the "properties", since we're already use that object for parsing other metadata

Reply:
I would argue we don't have much of a convention for storing collections in JSON. The only other things we store are the Atlas Objects and the flag properties. The Atlas Objects have to be in GeoJson convention. The properties we keep in as JSON objects as it fits well with the data being named fields in the flag. An important part of the suggestions is being able to pair them with a flagged object by the atlas id and item type, so I think it makes sense to gave it in an object with that identifier as the key. It also as close as you can get to a set in JSON because duplicate keys are not allowed. 
I did think about adding it to the properties, but thought it was distinct enough from the flag meta data. I also thought it might be easy for human reading if it was separate and not bloating the properties. However, I don't think this is a big deal either way, and in fact the geojson output has it in the feature properties. So if that is the place we think it fits best then I am happy to move it.

Reply:
One other option I considered was requiring a suggestion to have an associated flagged object and placing the suggestion in the features properties. This approach does not require the extra identifier, as it is already in the feature properties. I did not go with this because I wanted to keep the suggestions unrestricted for now, and I though it may cause issues with how we currently use the feature properties in post processing tools.

Reply:
In the case where we have a flag holding multiple osm objects (e.g. way 1, way 2.... picked up by a walker) wouldn't we want some sort of separation for the fix suggestions, as they only apply to distinct objects? Maybe an array might work better in this case, or some sort of suggestion merging during flag deduplication. Or your last idea might address this

Reply:
@seancoulter, this is exactly the case in the AreasWithHighwayTagCheck example included with this PR. My thought was that as we are flagging each edge we should have a suggestion for each edge, marked by the atlas ID. We are planning on handling the merging of the edges back into a way as part of the deserialization and MR upload process.

Reply:
Nice, sounds good